src/main/java/org/opendatasoft/elasticsearch/ingest/GeoExtensionProcessor.java

```diff
@@ -118,9 +119,27 @@ public IngestDocument execute(IngestDocument ingestDocument) throws IOException,
 
             // buildS4J() will try to clean up and fix the shape. If it fails, an exception is raised
             // Included fixes:
-            // - point deduplication
             // - dateline warping (enforce lon in [-180,180])
-            Shape shape = shapeBuilder.buildS4J();
+            Shape shape = null;
+            try {
+                try {
+                    shape = shapeBuilder.buildS4J();
+                } catch (InvalidShapeException e) {
+                    // buildS4J does not always deduplicate points
+                    if (e.getMessage().contains("duplicate")) {
```

Comment:
I'm not fan of getting the exception message to be tested in case the message change. What about not testing at all and trying to deduplicate when `InvalidShapeException` is raised ?